[Fix] logs http basic issue # 9755

wuxiansen opened the pull request:
Fix logs Basic issue
 
## Contribution Checklist
 
### Purpose of this pull request

This pull request supplements test cases for fixing basic log-related issues, ensuring that the logging functionality works correctly in scenarios where HTTP Basic authentication is enabled. By adding the RestApiHttpBasicTest test class, it verifies the correctness of log-related REST API responses with Basic authentication, including interface accessibility, response format, and content integrity, further ensuring the stability of the logging feature.


### Does this PR introduce _any_ user-facing change?

No. The added test cases are solely for internal functional verification and do not involve changes to user-visible behaviors, configuration methods, or output results. They have no impact on user workflows.


### How was this patch tested?

1. New test cases: The RestApiHttpBasicTest class was implemented to specifically test the behavior of log-related REST APIs when HTTP Basic authentication is enabled, including:
Verifying that the /overview endpoint returns a 200 OK status and contains necessary fields (e.g., projectVersion) with valid authentication.
Testing the /logs endpoint with different formats (e.g., JSON) to check for non-empty responses, correct status codes, and expected Content-Type headers.
Simulating job submission and validating that the log endpoint correctly returns job log information.

2. Test environment: A test cluster was set up using Hazelcast in local mode, with HTTP port and Basic authentication credentials (username admin, password admin) configured to match real-world scenarios.

3. Execution method: Tests were run using JUnit 5 to automatically verify interface behavior. All assertions passing indicate that the functionality works as expected.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)

Commit:
# 9755
Fix logs Basic issue

Commit:
Trigger workflow

wuxiansen commented:
@hailin0 @davidzollo  Hello, can you help me take a look at this PR?

Commit:
Trigger workflow

Commit:
# 9755
Modify code style

Commit:
# 9755
Modify DisabledOnOs OS.WINDOWS

wuxiansen commented:
> @hailin0 @davidzollo Hello, can you help me take a look at this PR?

I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail? 

PTAL 

<img width="2734" height="1446" alt="image" src="https://github.com/user-attachments/assets/ee253ba4-4135-4f56-9ff8-aab81a613788" />

<img width="2784" height="822" alt="image" src="https://github.com/user-attachments/assets/444e0560-7ba6-48c2-9e61-98a2acc4db87" />

dybyte commented:
> I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail?

Please pull the latest changes and try again.

Commit:
# 9755
Closure processing

Commit:
Merge remote-tracking branch 'upstream/dev' into feature-9755

wuxiansen commented:
> > I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail?
> 
> Please pull the latest changes and try again.

Executing the failures one by one can be done .

@liucongjy  @zhangshenghang 

`BaseServletTest.testWriteJsonWithObject` can pass the test locally, but why can't it pass in actions?

PTAL

<img width="1182" height="194" alt="image" src="https://github.com/user-attachments/assets/5ff2d7c1-e84f-424c-a539-b5dd18a1f3ee" />

<img width="1136" height="310" alt="image" src="https://github.com/user-attachments/assets/50d56201-4ccf-4fb2-896e-b00fd18806c9" />

<img width="1176" height="240" alt="image" src="https://github.com/user-attachments/assets/aa361f4f-3334-48aa-b0b8-2c656fcb4f25" />

```bash
[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    BaseServletTest.testWriteJsonWithObject:71->testLogRestApiResponse:147 expected: <200> but was: <401>
Error:  Errors: 
Error:    RestApiHttpsForTruststoreTest.testRestApiHttps:136 » IO Server returned HTTP r...
Error:    RestApiHttpsTest.testRestApiHttp:88 » IO Server returned HTTP response code: 4...
Error:    RestApiHttpsTest.testRestApiHttps:109 » IO Server returned HTTP response code:...
[INFO] 
Error:  Tests run: 96, Failures: 1, Errors: 3, Skipped: 6
[INFO] 

```

https://github.com/wuxiansen/seatunnel/actions/runs/18870236917/job/53957365698

Commit:
Trigger workflow

Commit:
# 9755
ConfigProvider.locateAndGetSeaTunnelConfig() single-case , Use case is reset later

wuxiansen commented:
> > > I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail?
> > 
> > 
> > Please pull the latest changes and try again.
> 
> Executing the failures one by one can be done .
> 
> @liucongjy @zhangshenghang
> 
> `BaseServletTest.testWriteJsonWithObject` can pass the test locally, but why can't it pass in actions?
> 
> PTAL
> 
> <img alt="image" width="1182" height="194" src="https://private-user-images.githubusercontent.com/6215311/507026868-5ff2d7c1-e84f-424c-a539-b5dd18a1f3ee.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE4Mzk5NTYsIm5iZiI6MTc2MTgzOTY1NiwicGF0aCI6Ii82MjE1MzExLzUwNzAyNjg2OC01ZmYyZDdjMS1lODRmLTQyNGMtYTUzOS1iNWRkMThhMWYzZWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMzBUMTU1NDE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2U0NGEyOTg5MDc3YWNmYjIwOTE3ZGY2MGM3YThlNDY0ZTUzNjYxZjJjZmFiM2JjNDc0YjdjYjIyOWI1OWNjNyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.EsIFO94huF5rEGTeHL1Vnrv-D_V1UpvuV8OB98mJ1Gg"> <img alt="image" width="1136" height="310" src="https://private-user-images.githubusercontent.com/6215311/507028909-50d56201-4ccf-4fb2-896e-b00fd18806c9.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE4Mzk5NTYsIm5iZiI6MTc2MTgzOTY1NiwicGF0aCI6Ii82MjE1MzExLzUwNzAyODkwOS01MGQ1NjIwMS00Y2NmLTRmYjItODk2ZS1iMDBmZDE4ODA2YzkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMzBUMTU1NDE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzQyMzY2MjIzMDExMzc3N2JkNGJhYTY3YmExYjY0YmFjNjkzYTMzMGZkM2IzZWQ3NDhmZjFiNDhhN2E2Y2RiMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.079wImcZpOMgF2MdvO8IDX-EOqXlOTScNnhKPaFTbwk"> <img alt="image" width="1176" height="240" src="https://private-user-images.githubusercontent.com/6215311/507029088-aa361f4f-3334-48aa-b0b8-2c656fcb4f25.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE4Mzk5NTYsIm5iZiI6MTc2MTgzOTY1NiwicGF0aCI6Ii82MjE1MzExLzUwNzAyOTA4OC1hYTM2MWY0Zi0zMzM0LTQ4YWEtYjBiOC0yYzY1NmZjYjRmMjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MTAzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTEwMzBUMTU1NDE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTA3ZWI3OThiOWYzNzU4MmZiMjFkZDM5MWQ4MzkwYjU3YjA5OTQ1MDExMjg5YzZjYjA4ZTAzYmU5MTg5ZmE0ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.EZW-ItfoHnA6pzVf4iWPP1Uz8ANHZ_n4t0ps15lzukc">
> ```shell
> [INFO] 
> [INFO] Results:
> [INFO] 
> Error:  Failures: 
> Error:    BaseServletTest.testWriteJsonWithObject:71->testLogRestApiResponse:147 expected: <200> but was: <401>
> Error:  Errors: 
> Error:    RestApiHttpsForTruststoreTest.testRestApiHttps:136 » IO Server returned HTTP r...
> Error:    RestApiHttpsTest.testRestApiHttp:88 » IO Server returned HTTP response code: 4...
> Error:    RestApiHttpsTest.testRestApiHttps:109 » IO Server returned HTTP response code:...
> [INFO] 
> Error:  Tests run: 96, Failures: 1, Errors: 3, Skipped: 6
> [INFO] 
> ```
> 
> https://github.com/wuxiansen/seatunnel/actions/runs/18870236917/job/53957365698


I seem to have found the problem. The setUP method is a bit confusing, making people think that each test class is independent, but in fact it is not. ConfigProvider.locateAndGetSeaTunnelConfig() single-case, unified management through org.apache.seatunnel.engine.common.config.YamlSeaTunnelDomConfigProcessor. as it happens
The newly added RestApiHttpBasicTest class is executed first, as can be seen from the following log.

```
Oct 30, 2025 5:21:59 AM com.hazelcast.internal.server.tcp.TcpServerConnection
INFO: [localhost]:5801 [runner_CoordinatorServiceTest_testCleanPendingJobMasterMap] [5.1] Connection[id=13, /127.0.0.1:5801->/127.0.0.1:37651, qualifier=null, endpoint=[localhost]:5803, remoteUuid=048f417e-3de9-4ff1-a253-e33d2ba479a8, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
org.apache.seatunnel.engine.common.loader.SeaTunnelChildFirstClassLoader@2759114
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.807 s - in org.apache.seatunnel.engine.server.rest.RestApiHttpBasicTest
[INFO] Running org.apache.seatunnel.engine.server.rest.RestApiHttpsForTruststoreTest
Oct 30, 2025 5:22:01 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading configuration '/home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml' from System property 'seatunnel.config'

...

Oct 30, 2025 5:22:02 AM com.hazelcast.internal.server.tcp.TcpServerConnection
INFO: [localhost]:5801 [runner_CoordinatorServiceTest_testCleanPendingJobMasterMap] [5.1] Connection[id=15, /127.0.0.1:5801->/127.0.0.1:43985, qualifier=null, endpoint=[localhost]:5804, remoteUuid=92d3d57e-c593-42cd-a4f6-6cea21527200, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
org.apache.seatunnel.engine.common.loader.SeaTunnelChildFirstClassLoader@2759114
Error:  Tests run: 4, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 2.586 s <<< FAILURE! - in org.apache.seatunnel.engine.server.rest.RestApiHttpsForTruststoreTest
Error:  testRestApiHttps  Time elapsed: 0.124 s  <<< ERROR!
java.io.IOException: Server returned HTTP response code: 401 for URL: https://localhost:18443/overview
	at org.apache.seatunnel.engine.server.rest.RestApiHttpsForTruststoreTest.testRestApiHttps(RestApiHttpsForTruststoreTest.java:136)

[INFO] Running org.apache.seatunnel.engine.server.rest.RestApiHttpsTest
Oct 30, 2025 5:22:03 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading configuration '/home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml' from System property 'seatunnel.config'

...

Oct 30, 2025 5:22:05 AM com.hazelcast.internal.server.tcp.TcpServerConnection
INFO: [localhost]:5801 [runner_CoordinatorServiceTest_testCleanPendingJobMasterMap] [5.1] Connection[id=17, /127.0.0.1:5801->/127.0.0.1:36841, qualifier=null, endpoint=[localhost]:5805, remoteUuid=6f488a61-6344-43da-9a6c-3070b4321d7e, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Connection closed by the other side
org.apache.seatunnel.engine.common.loader.SeaTunnelChildFirstClassLoader@2759114
Oct 30, 2025 5:22:05 AM org.apache.seatunnel.engine.server.CoordinatorService
INFO: [localhost]:5801 [runner_CoordinatorServiceTest_testCleanPendingJobMasterMap] [5.1] 
Error:  Tests run: 3, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 2.293 s <<< FAILURE! - in org.apache.seatunnel.engine.server.rest.RestApiHttpsTest
Error:  testRestApiHttps  Time elapsed: 0.036 s  <<< ERROR!
java.io.IOException: Server returned HTTP response code: 401 for URL: https://localhost:28443/overview
	at org.apache.seatunnel.engine.server.rest.RestApiHttpsTest.testRestApiHttps(RestApiHttpsTest.java:109)

Error:  testRestApiHttp  Time elapsed: 0.005 s  <<< ERROR!
java.io.IOException: Server returned HTTP response code: 401 for URL: http://localhost:28080/overview
	at org.apache.seatunnel.engine.server.rest.RestApiHttpsTest.testRestApiHttp(RestApiHttpsTest.java:88)

[INFO] Running org.apache.seatunnel.engine.server.rest.BaseServletTest
Oct 30, 2025 5:22:06 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading configuration '/home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml' from System property 'seatunnel.config'
Oct 30, 2025 5:22:06 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Using configuration file at /home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml
Oct 30, 2025 5:22:06 AM org.apache.seatunnel.engine.common.config.YamlSeaTunnelDomConfigProcessor
INFO: Dynamic slot is enabled, the schedule strategy is set to REJECT
Oct 30, 2025 5:22:06 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading 'hazelcast.yaml' from the classpath.
Oct 30, 2025 5:22:06 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading configuration '/home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml' from System property 'seatunnel.config'

...

Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 2.349 s <<< FAILURE! - in org.apache.seatunnel.engine.server.rest.BaseServletTest
Error:  testWriteJsonWithObject  Time elapsed: 0.027 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <200> but was: <401>
	at org.apache.seatunnel.engine.server.rest.BaseServletTest.testLogRestApiResponse(BaseServletTest.java:147)
	at org.apache.seatunnel.engine.server.rest.BaseServletTest.testWriteJsonWithObject(BaseServletTest.java:71)

[INFO] Running org.apache.seatunnel.engine.server.CoordinatorServiceWithCancelPendingJobTest
Oct 30, 2025 5:22:08 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading configuration '/home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml' from System property 'seatunnel.config'
Oct 30, 2025 5:22:08 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Using configuration file at /home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml
Oct 30, 2025 5:22:08 AM org.apache.seatunnel.engine.common.config.YamlSeaTunnelDomConfigProcessor
INFO: Dynamic slot is enabled, the schedule strategy is set to REJECT
Oct 30, 2025 5:22:08 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading 'hazelcast.yaml' from the classpath.
Oct 30, 2025 5:22:08 AM com.hazelcast.internal.config.AbstractConfigLocator
INFO: Loading configuration '/home/runner/work/seatunnel/seatunnel/seatunnel-engine/seatunnel-engine-server/src/test/resources/seatunnel.yaml' from System property 'seatunnel.config'
```

Local reproduction is reproduced by using the following command:
```bash
 mvn -pl seatunnel-engine/seatunnel-engine-server test -Dtest=RestApiHttpBasicTest,BaseServletTest,RestApiHttpsTest,RestApiHttpsForTruststoreTest -DfailIfNoTests=false
```

Solution: After setting `RestApiHttpBasicTest`, use `AfterAll` to clean up the Basic configuration.

PR

Commit:
Merge remote-tracking branch 'upstream/dev' into feature-9755

# Conflicts:
#	seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/LogService.java

wuxiansen commented:
> > I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail?
> 
> Please pull the latest changes and try again.

Use case execution is completed. I re-executed it several times, and the redis test case has a high probability of failure. I re-executed it many times, but luckily it finally worked. 

PTAL

wuxiansen commented:
> > > I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail?
> > 
> > 
> > Please pull the latest changes and try again.
> 
> Use case execution is completed. I re-executed it several times, and the redis test case has a high probability of failure. I re-executed it many times, but luckily it finally worked.
> 
> PTAL

I’ve been keeping an eye on it lately, and it doesn’t seem like there’s been much progress. If you have any ideas or think we need to make adjustments, we can talk it through together. 🤝

dybyte commented:
> > > > I only fix a certain piece of content, and the other module code has not changed. Why do module test cases that have not changed keep compiling fail?
> > > 
> > > 
> > > Please pull the latest changes and try again.
> > 
> > 
> > Use case execution is completed. I re-executed it several times, and the redis test case has a high probability of failure. I re-executed it many times, but luckily it finally worked.
> > PTAL
> 
> I’ve been keeping an eye on it lately, and it doesn’t seem like there’s been much progress. If you have any ideas or think we need to make adjustments, we can talk it through together. 🤝

I’ve opened a PR addressing the flaky Redis tests. #10024 
Thanks for reporting this issue!

dybyte commented:
cc @zhangshenghang @hawk9821 — could you please take a look when you have a moment?

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/LogService.java:
```diff
@@ -98,6 +108,40 @@ public List<Tuple3<String, String, String>> allLogNameList(String jobId) {
+        if (StringUtils.isNotBlank(basicUser) && StringUtils.isNotBlank(basicPass)) {
+            httpConfig.setBasicAuthUsername(basicUser);
+            httpConfig.setBasicAuthPassword(basicPass);
+        }
```
Could you explain why this is needed?

dybyte replied:
Could you explain why this is needed?

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java:
```diff
@@ -0,0 +1,208 @@
+    public void testRestApiOverview() throws Exception {
+        HttpURLConnection conn = null;
+        try {
+            java.net.URL url = new java.net.URL("http://localhost:" + HTTP_PORT + "/overview");
```
Let's import it.

wuxiansen replied:
OK

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java:
```diff
@@ -0,0 +1,208 @@
+        startJob();
+        HttpURLConnection conn = null;
+        try {
+            java.net.URL url =
```
ditto

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java:
```diff
@@ -0,0 +1,208 @@
+    public void testLogRestApiResponse(String format) throws IOException {
+        HttpURLConnection conn = null;
+        try {
+            java.net.URL url =
```
ditto

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/BaseLogService.java:
```diff
@@ -47,30 +47,87 @@ public String getLogPath() {
+                                .encodeToString(
+                                        (user + ":" + pass)
+                                                .getBytes(java.nio.charset.StandardCharsets.UTF_8));
+                connection.setRequestProperty("Authorization", "Basic " + token);
```
Let's reuse 
https://github.com/apache/seatunnel/blob/5681fe3bb7c7c5a99a275e2139e4d06b2d780b7b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/filter/BasicAuthFilter.java#L43-L44

wuxiansen replied:
OK

wuxiansen replied:
> Let's reuse
> 
> https://github.com/apache/seatunnel/blob/5681fe3bb7c7c5a99a275e2139e4d06b2d780b7b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/filter/BasicAuthFilter.java#L43-L44

This is private and can be referenced,`HttpHeaders.Authorization`.  BASIC_PREFIX feels like it can be defined in this test class.

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/LogService.java:
```diff
@@ -98,6 +108,40 @@ public List<Tuple3<String, String, String>> allLogNameList(String jobId) {
+            // Compatible with older versions: If HttpConfig does not have these methods, use system
+            // properties or environment variables to find out
+            basicUser = System.getProperty("seatunnel.http.user", System.getenv("BASIC_AUTH_USER"));
+            basicPass = System.getProperty("seatunnel.http.pass", System.getenv("BASIC_AUTH_PASS"));
```
Does this really provide backward compatibility?
As far as I know, older versions didn’t support this feature at all,
so this fallback might not actually be effective.

wuxiansen replied:
Yes, I’ve considered that point. If it doesn’t actually improve backward compatibility, I’m fine with removing this part.

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/BaseLogService.java:
```diff
@@ -47,30 +47,87 @@ public String getLogPath() {
+                        java.util.Base64.getEncoder()
+                                .encodeToString(
+                                        (user + ":" + pass)
+                                                .getBytes(java.nio.charset.StandardCharsets.UTF_8));
```
Let's import it.

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/BaseLogService.java:
```diff
@@ -47,30 +47,87 @@ public String getLogPath() {
+            while ((len = in.read(buf)) != -1) {
+                baos.write(buf, 0, len);
+            }
+            return baos.toString(java.nio.charset.StandardCharsets.UTF_8.name());
```
ditto

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/LogService.java:
```diff
@@ -70,15 +73,22 @@ public List<Tuple3<String, String, String>> allLogNameList(String jobId) {
+                                    ? sendGet(logUrl, result.basicUser, result.basicPass)
+                                    : sendGet(logUrl);
+
+                    if (allName == null || allName.trim().isEmpty()) {
```
Just wondering — was `StringUtils.isBlank()` intentionally replaced?

wuxiansen replied:
> Just wondering — was `StringUtils.isBlank()` intentionally replaced?

No, it wasn’t intentional. I had it in my original commit, but another contributor modified this section later, which caused a conflict. When resolving it, I noticed the logic was essentially the same, so I kept that version. The original intention was to avoid adding an additional dependency. It probably wasn’t very considerate to make that change without notifying others — thanks for pointing this out!  👍🏻

wuxiansen replied:
Removed HttpBasic object. Optimized ：
```java

                    String allName =
                            httpConfig.isEnableBasicAuth()
                                    ? sendGet(
                                            logUrl,
                                            httpConfig.getBasicAuthUsername(),
                                            httpConfig.getBasicAuthPassword())
                                    : sendGet(logUrl);

                    if (StringUtils.isBlank(allName)) {
                        log.warn(
                                "GET {} returned empty body (null/empty). Skip this node.", logUrl);
                        return;
                    }
```

dybyte reviewed (commented):
Thanks @wuxiansen

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/HttpBasic.java:
```diff
@@ -0,0 +1,26 @@
+public class HttpBasic {
+    public String basicUser;
+    public String basicPass;
+}
```
It seems this class duplicates `LogService.HttpBasic`.

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java:
```diff
@@ -0,0 +1,208 @@
+        HttpURLConnection conn = null;
+        try {
+            java.net.URL url =
+                    new java.net.URL("http://localhost:" + HTTP_PORT + "/logs?format=JSON");
```
I think we can reuse https://github.com/apache/seatunnel/blob/5681fe3bb7c7c5a99a275e2139e4d06b2d780b7b/seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestConstant.java#L86

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java:
```diff
@@ -0,0 +1,208 @@
+        HttpURLConnection conn = null;
+        try {
+            java.net.URL url =
+                    new java.net.URL("http://localhost:" + HTTP_PORT + "/logs?format=" + format);
```
ditto

wuxiansen replied:
All comments have been resolved. 🤝

Commit:
Code optimization
# 9755

Commit:
Merge remote-tracking branch 'upstream/dev' into feature-9755

Commit:
Code optimization
# 9755

Commit:
Code optimization
# 9755

Commit:
Code optimization
# 9755

dybyte commented on seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java:
```diff
@@ -0,0 +1,215 @@
+                //  "logName" : "job-${ctx:ST-JID}.log"
+                // } ]
+                String response = in.lines().collect(Collectors.joining());
+                Assertions.assertNotNull(response);
```
```suggestion
                Assertions.assertFalse(StringUtils.isBlank(response));
```

dybyte replied:
Maybe we need to import it. I forgot it.
```
import org.apache.seatunnel.shade.org.apache.commons.lang3.StringUtils;
```

wuxiansen replied:
Yep, it’s done.

wuxiansen committed:
Update seatunnel-engine/seatunnel-engine-server/src/test/java/org/apache/seatunnel/engine/server/rest/RestApiHttpBasicTest.java

Co-authored-by: dy102 <132787602+dybyte@users.noreply.github.com>

Commit:
import StringUtils
# 9755

dybyte reviewed (approved):
LGTM if CI passes. Thanks @wuxiansen

zhangshenghang commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/BaseLogService.java:
```diff
@@ -47,30 +53,87 @@ public String getLogPath() {
+     *
+     * @param urlString url
+     * @param user name
+     * @param pass password
+     * @return log
+     */
+    protected String sendGet(String urlString, String user, String pass) {
+        HttpURLConnection connection = null;
+        try {
+            connection = (HttpURLConnection) new URL(urlString).openConnection();
+            connection.setRequestMethod("GET");
+            connection.setConnectTimeout(5000);
+            connection.setReadTimeout(5000);
+
+            // Basic Auth
+            if (user != null && pass != null) {
+
+                Encoder encoder = Base64.getEncoder();
+                String auth = user + ":" + pass;
+                String token = encoder.encodeToString(auth.getBytes(StandardCharsets.UTF_8));
+
+                connection.setRequestProperty(AUTHORIZATION_HEADER, BASIC_PREFIX + token);
+            }
+
+            connection.connect();
+
+            int code = connection.getResponseCode();
+            if (connection.getResponseCode() == 200) {
+                return readAll(connection.getInputStream());
+            }
+            log.warn("GET {} -> HTTP {}", urlString, code);
+        } catch (IOException e) {
+            log.error("Send GET failed: url={}, err={}", urlString, ExceptionUtils.getMessage(e));
+        } finally {
+            if (connection != null) {
+                connection.disconnect();
+            }
+        }
+        return null;
+    }
```
It seems that a lot of their code is duplicated; let's merge it.

zhangshenghang commented on seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/service/BaseLogService.java:
```diff
@@ -47,30 +53,87 @@ public String getLogPath() {
-                    return baos.toString();
-                }
+                return readAll(connection.getInputStream());
             }
```
When code != 200, the resource is not released; please release it.
You can refer to:
```java
int code = connection.getResponseCode();
          if (code == 200) {
              return readAll(connection.getInputStream());
          }

          try (InputStream errorStream = connection.getErrorStream()) {
              if (errorStream != null) {
                  byte[] buffer = new byte[1024];
                  while (errorStream.read(buffer) != -1) {
                  }
              }
          }
```

wuxiansen replied:
all done.

Commit:
Optimize sendGet method
# 9755

zhangshenghang reviewed (approved):
LGTM， If CI pass
